pkg/archive/archive_windows.go

```diff
@@ -31,14 +31,11 @@ func CanonicalTarNameForPath(p string) string {
 // chmodTarEntry is used to adjust the file permissions used in tar header based
 // on the platform the archival is done.
 func chmodTarEntry(perm os.FileMode) os.FileMode {
-	// perm &= 0755 // this 0-ed out tar flags (like link, regular file, directory marker etc.)
-	permPart := perm & os.ModePerm
-	noPermPart := perm &^ os.ModePerm
-	// Add the x bit: make everything +x from windows
-	permPart |= 0111
-	permPart &= 0755
+	// Remove group- and world-writable bits.
+	perm &= 0o755
 
-	return noPermPart | permPart
+	// Add the x bit: make everything +x on Windows
+	return perm | 0o111
```

Comment:
1. Commit 1a22418f9f1573ca8521831d52c8e0562cb3ef8f (https://github.com/moby/moby/pull/11148) changed permissions to `0700`
   on Windows, or more factually, it removed `rw` (`chmod g-rw,o-rw`) and added
   executable bits (`chmod u+x`) to address https://github.com/moby/moby/issues/11047.
2. This was too restrictive, and b7dc9040f04fe8dacd2f14c6b93d1b0bb6bde333 (https://github.com/moby/moby/pull/11395) changed
   permissions to only remove the group- and world-writable bits to give read and
  execute access to everyone, but setting execute permissions for everyone to address https://github.com/moby/moby/issues/11246
3. However, this also removed the non-permission bits, so 41eb61d5c2a44c745b11ada948034fd3bcc42db5 (https://github.com/moby/moby/pull/32959)
   updated the code to preserve those, and keep parity with Linux.

This changes it back to `2.`. I wonder (_think_) _permission_ bits (read, write)
can be portable, except for the _executable_ bit (which is not present on Windows).
The alternative could be to keep the permission bits, and only set the executable
bit (`perm | 0111`) for everyone (equivalent of `chmod +x`), but that likely would
be a breaking change.